modules/personalize_taxonomy_context/plugins/visitor_context/TaxonomyContext.inc

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/**
+ * @file
+ * Provides a visitor context plugin for taxonomy.
+
+ */
+
+class TaxonomyContext extends PersonalizeContextBase {
+  /**
+   * Implements PersonalizeContextInterface::create().
+   */
+  public static function create(PersonalizeAgentInterface $agent = NULL, $selected_context = array()) {
+    return new self($agent, $selected_context);
+  }
+
+  /**
+   * Implements PersonalizeContextInterface::getOptions().
+   */
+  public static function getOptions() {
+    $vocabularies = taxonomy_get_vocabularies();
+    $options = array();
+    foreach ($vocabularies as $vocabulary) {
+      $machine_name = $vocabulary->machine_name;
+      $options[$machine_name] = array(
+        'name' => $vocabulary->name,
+        'group' => 'Taxonomy',
+        'id' => $vocabulary->vid,
+      );
+    }
+    return $options;
+  }
+
+  /**
+   * Implements PersonalizeContextInterface::getPossibleValues().
+   */
+  public function getPossibleValues($limit = FALSE) {
+    $possible_values = array();
+    $options = $this->getOptions();
+    $free_tagging_vocabulary_machine_names = SELF::getFreeTaggingVocabulariesMachineNames();
```

Comment:
I'm saying that free-tagging versus non-free-tagging is not a valid distinction to make about vocabularies - it used to be (in Drupal 6) but not anymore - I was wrong to suggest we could make this distinction. I think the best we can do here is pick a number n and offer the first n tags for each vocabulary as possible values to use for fixed targeting. I'm not sure how to pick n - maybe 50?